pkg/providers/vsphere/validator.go

```diff
@@ -385,3 +423,184 @@ func (v *Validator) validateControlPlaneIpUniqueness(spec *Spec) error {
 	}
 	return nil
 }
+
+func (v *Validator) validateUserPrivs(ctx context.Context, spec *Spec, vuc *VSphereUserConfig) error {
+	controlPlaneMachineConfig := spec.controlPlaneMachineConfig()
+
+	privObjs := []map[string]string{
+		// validate global root priv settings are correct
+		{
+			"type":         vsphereTypeFolder,
+			"privsContent": vsphereGlobalPrivsFile,
+			"path":         vsphereRootPath,
+		},
+
+		// validate object-level priv settings are correct
+		{
+			"type":         vsphereTypeDatastore,
+			"privsContent": vsphereUserPrivsFile,
+			"path":         controlPlaneMachineConfig.Spec.Datastore,
+		},
+		{
+			"type":         vsphereTypeResourcePool,
+			"privsContent": vsphereUserPrivsFile,
+			"path":         controlPlaneMachineConfig.Spec.ResourcePool,
+		},
+		{
+			"type":         vsphereTypeNetwork,
+			"privsContent": vsphereUserPrivsFile,
+			"path":         spec.datacenterConfig.Spec.Network,
+		},
+
+		// validate Administrator role (all privs) on VM folder and Template folder
+		{
+			"type":         vsphereTypeFolder,
+			"privsContent": vsphereAdminPrivsFile,
+			"path":         controlPlaneMachineConfig.Spec.Folder,
+		},
+		// Not implementing a check on the template because I'm not sure if we always require/provide/allow template permissions
```

Comment:
I don't understand what this section means:
https://anywhere.eks.amazonaws.com/docs/reference/vsphere/vsphere-preparation/#deploy-an-ova-template

Does it mean that a user doesn't necessarily have to have access to the templates folder?

Reply:
This validation can be conditional. If the template field in vsphere config object has been configured, we can *skip* the validation.

If we *do* apply the validation, we need to do so on:
- control plane template directory
- worker node template directory
- external etcd directory